content/guides/hot-module-replacement.md

```diff
@@ -18,140 +19,271 @@ related:
+
+This is happening because the button's `onclick` event handler is still bound to the original `printMe` function.
 
-``` js
-class Logger {
-  log(text) {
-    console.log('Logging some text: ', text)
+To make this work with HMR we need to update that binding to the new `printMe` function using `module.hot.accept`
+
+__print.js__
+
+``` diff
+  import _ from 'lodash';
+  import printMe from './print.js';
+
+  if (module.hot) {
+    module.hot.accept('./print.js', function() {
+      console.log('Accepting the updated printMe module!');
+-     printMe();
++     document.body.removeChild(element);
++     element = component(); // Re-render the "component" to update the click handler
++     document.body.appendChild(element);
+    })
   }
-}
-```
 
-Even if the underlying module containing this class is patched with new code, any existing instances of the class still have the old `log` method. Meaning if we changed what that method does, it wouldn't be reflected in those old instances unless we re-instantiate them somehow using `module.hot.accept`.
+  let element = component();
 
-This is just one example, but there are many others that can easily trip people up. Luckily, there are a lot of loaders out there, some mentioned below, that will make using this process much easier.
+  function component() {
+    var element = document.createElement('div');
+    var btn = document.createElement('button');
 
+    element.innerHTML = _.join(['Hello', 'webpack'], ' ');
 
-## HMR with Stylesheets
+    btn.innerHTML = 'Click me and check the console!';
+    btn.onclick = printMe;  // onclick event is bind to the original printMe function
 
-We can use the `style-loader` to achieve Hot Module Replacement with CSS. This loader uses `module.hot.accept` behind the scenes to patch `<style>` tags when CSS dependencies are updated. So, with the following webpack configuration...
-
-``` js
-module.exports = {
-  // ...
-  module: {
-    rules: [
-      {
-        test: /\.css$/,
-        use: [ 'style-loader', 'css-loader' ]
-      }
-    ]
-  },
-  // ...
-}
+    element.appendChild(btn);
+
+    return element;
+  }
+
+  document.body.appendChild(element);
 ```
 
-hot loading stylesheets is a breeze...
+This is just one example, but there are many others that can easily trip people up. Luckily, there are a lot of loaders out there (some of which are mentioned below) that will make hot module replacement much easier.
 
-__index.js__
 
-``` js
-import Lib from './library';
-import './styles.css';
+## HMR with Stylesheets
+
+Hot Module Replacement with CSS is actually fairly straightforward with the help of the `style-loader`. This loader uses `module.hot.accept` behind the scenes to patch `<style>` tags when CSS dependencies are updated.
+
+First let's install both loaders with the following command: `npm install --save-dev style-loader css-loader`
```

Comment:
You should break the command to a new line e.g.

```` md
First let's install both loaders with the following command:

``` bash
npm install --save-dev style-loader css-loader
```
````